common/src/main/java/com/wynntils/features/ui/CustomTerritoryManagementScreenFeature.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright © Wynntils 2024.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.ui;
+
+import com.wynntils.core.components.Models;
+import com.wynntils.core.consumers.features.Feature;
+import com.wynntils.core.persisted.Persisted;
+import com.wynntils.core.persisted.config.Category;
+import com.wynntils.core.persisted.config.Config;
+import com.wynntils.core.persisted.config.ConfigCategory;
+import com.wynntils.core.persisted.storage.Storage;
+import com.wynntils.core.text.StyledText;
+import com.wynntils.handlers.wrappedscreen.event.WrappedScreenOpenEvent;
+import com.wynntils.mc.event.ContainerClickEvent;
+import com.wynntils.mc.event.MenuEvent;
+import com.wynntils.mc.event.ScreenClosedEvent;
+import com.wynntils.models.containers.containers.GuildTerritoriesContainer;
+import com.wynntils.screens.territorymanagement.TerritoryManagementScreen;
+import com.wynntils.utils.mc.KeyboardUtils;
+import com.wynntils.utils.mc.McUtils;
+import com.wynntils.utils.wynn.ContainerUtils;
+import java.util.regex.Pattern;
+import net.minecraft.world.inventory.AbstractContainerMenu;
+import net.minecraft.world.item.ItemStack;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+@ConfigCategory(Category.UI)
+public class CustomTerritoryManagementScreenFeature extends Feature {
+    private static final Pattern TERRITORY_MANAGE_ITEM_PATTERN = Pattern.compile("§e§lTerritories \\[.+\\]");
+    private static final Pattern MANAGE_TITLE_PATTERN = Pattern.compile(".+: Manage");
+    private static final Pattern BACK_BUTTON_PATTERN = Pattern.compile("§7§lBack");
+    private static final int TERRITORY_MANAGEMENT_SLOT = 14;
+
+    @Persisted
+    private final Config<ShiftBehavior> shiftBehaviorConfig = new Config<>(ShiftBehavior.DISABLED_IF_SHIFT_HELD);
+
+    @Persisted
+    public final Storage<Boolean> screenHighlightLegend = new Storage<>(true);
+
+    private boolean shiftClickedTerritoryItem = false;
+
+    private boolean customScreenOpened = false;
+    private boolean openTerritoryManagement = false;
+
+    @SubscribeEvent
+    public void onWrappedScreenOpen(WrappedScreenOpenEvent event) {
+        if (event.getWrappedScreenClass() != TerritoryManagementScreen.class) return;
+
+        switch (shiftBehaviorConfig.get()) {
+            case NONE -> {
+                event.setOpenScreen(true);
+                customScreenOpened = true;
+            }
+            case ENABLED_IF_SHIFT_HELD -> {
+                if (shiftClickedTerritoryItem) {
+                    event.setOpenScreen(true);
+                    customScreenOpened = true;
+                }
+            }
+            case DISABLED_IF_SHIFT_HELD -> {
+                if (!shiftClickedTerritoryItem) {
+                    event.setOpenScreen(true);
+                    customScreenOpened = true;
+                }
+            }
+        }
```

Comment:
Since they're all setting to true, maybe make a `shouldOpenScreen` from the switch and then set it based on that. Something like

```
boolean shouldOpenScreen = switch (shiftBehaviorConfig.get()) {
    case ENABLED_IF_SHIFT_HELD -> shiftClickedTerritoryItem;
    case DISABLED_IF_SHIFT_HELD -> !shiftClickedTerritoryItem;
    default -> true;
};

event.setOpenScreen(shouldOpenScreen);
customScreenOpened = shouldOpenScreen;
```

Reply:
Sure. Although the trade market screen does the same exact thing, I might as well change it.